### ISSUE TITLE
0 rows updated when that shoud have been 1

### ISSUE THREAD

**Issue by CraftedRO** [HeidiSQL/HeidiSQL#2051]
### Description

heidi seems like sometimes it does generate wrong update query when you update multiple values separately.

Well I created a video for better understanding of thie issue I have: https://youtu.be/Vrs7pa4Ajko

### HeidiSQL version

12.8.0.6956 (.zip one)

### Database server version

MariaDB 10.4.32

### Reproduction recipe

Video.

**Comment by ansgarbecker** [comment]
Please post the `CREATE TABLE` code here.

**Comment by CraftedRO** [comment]
https://github.com/TrinityCore/TrinityCore/blob/b7471244adb08f112214634b27d7daf2da19aa0f/sql/base/auth_database.sql#L69

**Comment by CraftedRO** [comment]
looks like this is not yet completely solved, I managed to reproduce it again on another table:
```sql
DROP TABLE IF EXISTS `spell_script_names`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `spell_script_names` (
  `spell_id` int NOT NULL,
  `ScriptName` char(64) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
  UNIQUE KEY `spell_id` (`spell_id`,`ScriptName`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;
```

**Comment by ansgarbecker** [comment]
Works fine here, on MariaDB 11.1. What's the problem exactly? I see the columns don't have a default value.

**Comment by CraftedRO** [comment]
yeah, interesting is that with the separate create table works, but with full db imported when I duplicate the row without keys (ctrl + ins) then in the row created I enter 1 and 1 for example, the error pops again.

**Comment by CraftedRO** [comment]
new video uploaded here https://youtu.be/mtE6EjQ_mC8

**Comment by ansgarbecker** [comment]
Ah, now I see you were using the "duplicate row without keys" item for that... Please note this also happens with any Insert item, not only with the "duplicate..." items. Caused by a NULL which is silently modified by the server.
Even with a similar MariaDB version (10.3.12 here) I get an SQL _error_ (not a warning as on your side) when trying to post the new row with a NULL in ScriptName. A warning does not block the insertion.
I cannot fix that in HeidiSQL, as with just a warning, HeidiSQL does not know there is a non-NULL value silently inserted by the server in `ScriptName`. Reason is that your server's `sql_mode` variable is missing "STRICT_TRANS_TABLES" - please add that, or just pay attention to such warnings. Alternatively, refresh the grid after inserting NULLs.
